machine-runner/tests/esm/runner.test.ts

```diff
@@ -588,6 +588,66 @@ describe('machine as async generator', () => {
     expect(toggleCount).toBe(2)
   })
 
+  describe('actual', () => {
+    it('should immediately resolve when caught up', async () => {
+      const r1 = new Runner(On, { toggleCount: 0 })
+      const machine = r1.machine
+      r1.feed([], true)
+      const value = (await machine.actual()).value
+      expect(value).not.toBe(null)
+    })
+
+    it('should resolve when commandLock is unlocked', async () => {
+      const r1 = new Runner(On, { toggleCount: 0 })
+      const machine = r1.machine
+      r1.feed([], true)
+
+      await r1.toggleCommandDelay({ delaying: true })
+      const commandsWhenOn = (await machine.actual()).value?.as(On)?.commands()
+      if (!commandsWhenOn) throw new Unreachable()
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      commandsWhenOn.toggle().catch(() => {})
+
+      let resolved = false
+      const actualPromise = machine.actual()
+      actualPromise.finally(() => (resolved = true))
+      expect(resolved).toBe(false)
```

Comment:
I know I checked this and we discussed it, but I don’t trust the JS ecosystem to maintain the property that the `.finally` would run immediately on a resolved Promise. Just for paranoia’s sake, would it hurt to sleep for 1ms between these lines?

Reply:
will do

Reply:
after reviewing this again, no. `finally` will be processed after the call to `toggleCommandDelay`, which, after the call there is already a sleep

Reply:
Sure, if `machine.actual()` works as intended. But if it erroneously were to return a resolved promise then `actualPromise.finally(...)` might perform its effects _after_ the `expect(resolved)` line.

This is what the ECMAScript standard has to say about this topic: the [semantics of Promise.prototype.then](https://tc39.es/ecma262/multipage/control-abstraction-objects.html#sec-performpromisethen) specify to use [HostEnqueuePromiseJob](https://tc39.es/ecma262/multipage/executable-code-and-execution-contexts.html#sec-hostenqueuepromisejob) to execute any continuations (including `.finally()`), which _does not_ specify that the callback is run immediately. It only requires that callbacks are executed in the order in which they are enqueued.

Reply:
https://github.com/Actyx/machines/pull/89 takes care of this